docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,358 @@
+# ADR-0008: Digest Calculation & Signing/Pinning (OCI Cosign Support)
+
+- **Status:** Proposed
+- **Deciders:** OCM Maintainers
+- **Date:** 2025-08-27
+- **Related:** OCM signing specification and examples
+
+---
+
+## Context
+
+Component Descriptors (CDs) are signed without altering digest fields during signing. Digests for resources, component references, and the component version are produced in a dedicated step and embedded into the descriptor. Signing uses fixed canonicalization (JCS, RFC 8785) over the descriptor to compute the component-version digest and create a signature envelope. When a component version references another, the digest of the referenced descriptor is embedded in the parent’s `references[]` and contributes to the parent’s component-version digest.
+
+Some workflows require an additional **OCI Cosign** signature over the same canonical payload, optionally uploaded to an OCI registry and recorded in a transparency log (Rekor).
+
+---
+
+## Decision
+
+### Two-step flow
+
+1. **Digest calculation** mutates the descriptor to embed required digests (resources, component references, component-version).
+2. **Signing** appends a signature envelope without computing or embedding digests. Optionally, the same payload may be **cosigned**.
+
+### Commands (verb–noun, long options only, no stdin)
+
+```bash
+# Step 1 — Digest calculation (mutates descriptor: digests only)
+ocm add digests cv <ref> --recurse --force
+
+# Step 2 — Signing (no digest mutation)
+# Direct signing via a single --cert (+ optional password).
+# Optional OCI Cosign signing for the same canonical payload.
+ocm sign cv <ref> \
+  --sig <slot> \
+  --pin <sha256:...> \
+  --cert <path> \
+  --password <pw> \
+  --password-prompt \
+  --cosign \
+  --cosign-upload \
+  --cosign-skip-tlog \
+  --cosign-annotation <key=value> \
+  --cosign-annotation <key=value> \
+  --cosign-identity-token <token-or-path>
+```
+
+- `<ref>`: descriptor reference (file path or repository reference).
+- `--recurse`: calculate digests for referenced component versions.
```

Comment:
I dont understand why we would need this if we already have an add digests command

Reply:
but we don't have it right? I thought I add this to the adr for completeness

Reply:
removed from adr